Amplify/Categories/Storage/StorageCategory+ClientBehavior.swift

```diff
@@ -81,4 +81,12 @@ extension StorageCategory: StorageCategoryBehavior {
                      resultListener: StorageListOperation.ResultListener?) -> StorageListOperation {
         return plugin.list(options: options, resultListener: resultListener)
     }
+
+    // MARK: - Async API -
+
+    @discardableResult
```

Comment:
the reason why the callback apis had the `@discardableResult` was because the result was returned on the callback listener, and most use cases do not care about the operation. Do we want to have `@discardableResult` for these async APIs as well? The fast operation return type is the URL that the developer cares about. I think it make sense to warn them if they are not doing anything with the result, forcing them to explicitly putting down `_ = Amplify.Storage.getURL...` if they really don't care about the return value

Reply:
We still support the Hub so that may be used instead of the return value so I think it will be alright to leave this attribute.

Reply:
okay sounds good